src/Microsoft.Health.Dicom.Core/Features/Query/QueryParser.cs

```diff
@@ -156,18 +157,22 @@ private static IReadOnlyCollection<QueryTag> GetQualifiedQueryTags(IReadOnlyColl
             KeyValuePair<string, string> queryParameter,
             IEnumerable<QueryTag> queryTags,
             bool fuzzyMatching,
+            QueryResource resourceType,
             out QueryFilterCondition condition)
         {
             condition = null;
 
+            bool isWorkitemInstance = resourceType == QueryResource.WorkitemInstances;
```

Comment:
There is lot of if and else going to find the right data based on ResourceType.
Instead of doing that in each method you can make 

QueryParser take in a interface that is implemented for different resource type and QueryParser remains agnostic to ResourceType

Reply:
Made into a abstract class and now the code path is different.